src/hid_unix.c

```diff
@@ -66,10 +66,12 @@ fido_hid_unix_wait(int fd, int ms, const fido_sigset_t *sigmask)
 		ts.tv_nsec = (ms % 1000) * 1000000;
 	}
 
-	if ((r = ppoll(&pfd, 1, ms > -1 ? &ts : NULL, sigmask)) < 1) {
-		if (r == -1)
-			fido_log_error(errno, "%s: ppoll", __func__);
-		return (-1);
+	do {
+		r = ppoll(&pfd, 1, ms > -1 ? &ts : NULL, sigmask);
+	} while (r == -1 && errno == EINTR);
+
+	if (r == -1) {
+		fido_log_error(errno, "%s: ppoll", __func__);
 	}
```

Comment:
I think this is broken if `ppoll` exits with `r == -1` and `errno != EINTR`, since the current code will then return 0, although the previous code returned -1 or not? Also it seems like for `r = 0` the behavior is also different, the previous code again returns -1, but your new code returns 0.

Something like this should work:
```suggestion
        if (r < 1) {
		if (r == -1)
			fido_log_error(errno, "%s: ppoll", __func__);
		return (-1);
        }
```

Reply:
Yes, you're right on both accounts. Good that you saw that `r == 0` case... I would never have found that :sweat_smile: 

I changed to code to return -1 if `r < 0` or `r == 0`. For the `r == 0` case, which means the timeout (if one was specified)  expired, I have also added a log statement. `fido_log_error` requires and errno code and since I do not really have one I used `ETIME` which means `Timer expired`. Not sure if that is a good idea :grimacing:

Reply:
I'm also fine with your changes, but wonder why you didn't take the original suggestion, which mimics the upstream code. `ETIME` does not really explain the error, but I don't really care much honestly, since the error message will be concrete enough.  Are we planning to upstream this eventually?

Reply:
For some reason I did not see the suggestion before... Maybe because of the missing write permissions :thinking: 

Yes, I think we should try to upstream this. Probably easier with your suggestion. Can you suggest it again :grimacing: ?

Reply:
Couldn't suggest in this thread, started a new one.